core/src/data/literal.rs

```diff
@@ -96,31 +96,27 @@ impl<T: AsRef<str>> PartialOrd<T> for Literal<'_> {
     }
 }
 
-impl PartialEq<Literal<'_>> for Literal<'_> {
-    fn eq(&self, other: &Literal) -> bool {
-        match (self, other) {
-            (Boolean(l), Boolean(r)) => l == r,
-            (Number(l), Number(r)) => l == r,
-            (Text(l), Text(r)) => l == r,
-            (Bytea(l), Bytea(r)) => l == r,
-            _ => false,
-        }
-    }
-}
-
-impl PartialOrd<Literal<'_>> for Literal<'_> {
-    fn partial_cmp(&self, other: &Literal) -> Option<Ordering> {
+impl<'a> PartialOrd<Literal<'a>> for Literal<'a> {
+    fn partial_cmp(&self, other: &Literal<'a>) -> Option<Ordering> {
         match (self, other) {
             (Boolean(l), Boolean(r)) => Some(l.cmp(r)),
             (Number(l), Number(r)) => Some(l.cmp(r)),
             (Text(l), Text(r)) => Some(l.cmp(r)),
             (Bytea(l), Bytea(r)) => Some(l.cmp(r)),
+            (Null, Null) => Some(Ordering::Equal),
             _ => None,
         }
     }
 }
 
 impl<'a> Literal<'a> {
+    pub fn evaluate_eq(&self, other: &Literal<'_>) -> bool {
+        match (self, other) {
+            (Null, Null) => true,
```

Comment:
The `evaluate_eq` should have the same implementation as the previous `PartialEq`.

```suggestion
            (Null, Null) => false,
```